### PR TITLE
feat(css): emit css asset

### PR DESCRIPTION
**Pull request by hyf0**

### CODE REVIEW

**Comment by hyf0** [comment]
* **#2253** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2253?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2251** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2251?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2250** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2250?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2248** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2248?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2247** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2247?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @hyf0 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by hyf0** [comment]
### Merge activity

* **Sep 17, 9:17 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/rolldown/rolldown/2250) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

File: crates/rolldown/src/ecmascript/ecma_module_factory.rs [hunk 1]
```diff
@@ -73,7 +74,7 @@ impl ModuleFactory for EcmaModuleFactory {
       &stable_id,
       ctx.options,
       &ctx.module_type,
-      args.source,
+      args.source.clone(),
```
Comment (IWANABETHATGUY) [inline crates/rolldown/src/ecmascript/ecma_module_factory.rs:77]: Clone the source maybe not a good idea, but you could merge it first. Maybe this is the  reason performance regressed in #2251

Comment (hyf0) [reply crates/rolldown/src/ecmascript/ecma_module_factory.rs:77]: Yeah. Let's merge it first. I have a way to solve it in later PRs.

**Review by IWANABETHATGUY — APPROVED** [review]

**Pull request merged**